File: server/src/graphs/nodes/rag-node.js
import { ragChainWithSourcesAndRerank } from "../../chains/rag-chain.js";

export const ragNode = async (state) => {
  const { userInput } = state;
  try {
    // T20: 使用重排序 + 来源 RAG Chain
    const { answer, sources } = await ragChainWithSourcesAndRerank.invoke({
      question: userInput,
    });
    return { ragResult: answer, ragSources: sources };
  } catch (err) {
    console.error("[ragNode]", err.message);
    return { ragResult: "查询知识库时出错", ragSources: [] };
  }
};

File: backend_py/graphs/nodes/rag_node.py
"""
RAG 检索节点
对照 JS: ragNode
"""
from chains.rag_chain import rag_chain


async def rag_node(state):
    user_input = state["userInput"]
    try:
        result = await rag_chain.ainvoke({"question": user_input})
        return {"ragResult": result}
    except Exception as e:
        print(f"[ragNode] {e}")
        return {"ragResult": "查询知识库时出错"}